app/services/katello/pulp3/repository/yum.rb

```diff
@@ -332,28 +344,64 @@ def copy_content_from_mapping(repo_id_map, options = {})
           multi_copy_units(repo_id_map, dependency_solving)
         end
 
-        def copy_content_for_source(source_repository, options = {}) # rubocop:disable Metrics/CyclomaticComplexity
-          filters = [ContentViewErratumFilter, ContentViewPackageGroupFilter, ContentViewPackageFilter].collect do |filter_class|
+        def filtered_errata(errata_filters, source_repository_ids)
+          final_errata = []
+          all_excluded_errata = []
+          unless errata_filters.blacklist.empty?
+            if errata_filters.whitelist.empty?
+              all_included_errata = ::Katello::Erratum.joins("inner join katello_repository_errata on katello_errata.id = katello_repository_errata.erratum_id").
+                joins("inner join katello_repositories on katello_repositories.id = katello_repository_errata.repository_id").
+                where("katello_repositories.id in (?)", source_repository_ids)
+            else
+              all_included_errata = errata_filters.whitelist.collect do |filter|
+                ::Katello::Erratum.joins("inner join katello_repository_errata on katello_errata.id = katello_repository_errata.erratum_id").
+                  where(filter.generate_clauses(nil)).where("katello_repository_errata.repository_id in (?)", source_repository_ids)
+              end
+              all_included_errata.flatten!
+            end
+            all_excluded_errata = errata_filters.blacklist.collect do |filter|
+              ::Katello::Erratum.joins("inner join katello_repository_errata on katello_errata.id = katello_repository_errata.erratum_id").
+                where(filter.generate_clauses(nil)).where("katello_repository_errata.repository_id in (?)", source_repository_ids)
+            end
+            all_excluded_errata.flatten!
+          end
+          return all_included_errata, all_excluded_errata
```

Comment:
They're not, because they're not needed if there isn't a blacklist.   I'm still trying to think how to word them to make it more clear.  I realize that `all_included_errata` is pretty useless outside this method because it's `[]` if there aren't exclusion filters.

Reply:
Okay, so I updated this method to always return both in case it'll be useful later. This however means that `final_filtered_errata` will be blank if there are no exclude filters.  Makes it a little confusing still, but better.  Really what it is is a list of errata that should not be excluded, but I can't think of a good variable name.  `override_inclusion_errata`?

Reply:
> Okay, so I updated this method to always return both in case it'll be useful later. 
> This however means that `final_filtered_errata` will be blank if there are no exclude filters. 
I think this makes sense, it should be blank if nothing is excluded.

> Makes it a little confusing still, but better. Really what it is is a list of errata that should not be excluded, but I can't think of a good variable name. `override_inclusion_errata`?
Maybe 'additional_included_errata' ?

Reply:
I think that would work well, I'll try that.